### ISSUE TITLE
False positive for `MultilineMethodCallIndentation` inside `[]`

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop-hq/rubocop#7726]
There is false positive for `MultilineMethodCallIndentation` inside `[]` (code as key).

(Probably related to #7338)

--------

## Expected behavior

Without offenses.

## Actual behavior

With offenses.

## Steps to reproduce the problem

```yaml
# .rubocop.yml
Layout/MultilineMethodCallIndentation:
  EnforcedStyle: indented
```

```ruby
# test.rb

# frozen_string_literal: true

foo[
  bar
    .baz
]
```

```
Inspecting 1 file
C

Offenses:

test.rb:5:5: C: Layout/MultilineMethodCallIndentation: Align .baz with bar on line 4.
    .baz
    ^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```
$ bundle exec rubocop -V
0.80.0 (using Parser 2.7.0.2, running on ruby 2.7.0 x86_64-linux)
```

**Comment by AlexWayfer** [comment]
It's still actual.
